atc/db/worker_artifact.go

```diff
@@ -0,0 +1,119 @@
+package db
+
+import (
+	"database/sql"
+	"errors"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/concourse/concourse/atc"
+	"github.com/lib/pq"
+)
+
+//go:generate counterfeiter . WorkerArtifact
+
+type WorkerArtifact interface {
+	ID() int
+	Name() string
+	BuildID() int
+	CreatedAt() time.Time
+	Volume(teamID int) (CreatedVolume, bool, error)
+}
+
+type artifact struct {
+	conn Conn
+
+	id        int
+	name      string
+	buildID   int
+	createdAt time.Time
+}
+
+func (a *artifact) ID() int              { return a.id }
+func (a *artifact) Name() string         { return a.name }
+func (a *artifact) BuildID() int         { return a.buildID }
+func (a *artifact) CreatedAt() time.Time { return a.createdAt }
+
+func (a *artifact) Volume(teamID int) (CreatedVolume, bool, error) {
```

Comment:
surprised this is given a team ID - will it ever have multiple volumes for different teams? or is this just for checking access?

Reply:
yeah its just for checking access. its really unfortunate that teamID exists on the volume itself. It would be a lot nicer and more intuitive if each "higher level resource" had the teamID.